etna/whatson/models.py

```diff
@@ -685,6 +685,34 @@ def primary_access_type(self):
         if primary_access := self.event_access_types.first():
             return primary_access.access_type
 
+    @cached_property
+    def date_time_range(self):
+        delta = self.end_date - self.start_date
+        # One session on one date where start and end times are the same
+        # return eg. Monday 1 January 2024, 19:00
+        if (self.start_date == self.end_date) and (len(self.sessions.all()) == 1):
+            return self.start_date.strftime("%A %-d %B %Y, %H:%M")
+        # One session on one date where there are values for both start time and end time
+        # eg. Monday 1 January 2024, 19:00–20:00 (note this uses an en dash)
+        if (delta.days == 0 and delta.seconds > 0) and (len(self.sessions.all()) == 1):
```

Comment:
Suggestion: `datetime` has `date` and `time` methods. Comparing `self.start_date.date() == self.end_date.date() and ... ( time not same )` might be clearer than using a `timedelta`.

Question: is there validation that enforces that events start before they end? If not, this might result in a negative delta which would cause this comparison to fail.

Reply:
Unfortunately there is no validation to stop events having an end date before the start date - it feels like there should be though, so is that actually a separate problem to fix?

Reply:
`.date()` and `.time()` methods used instead of `delta` in https://github.com/nationalarchives/ds-wagtail/pull/1470/commits/1238a0ca1875a7be135bf6da42d4f00d7d83375b